server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -72,11 +72,13 @@ func (b *BuildEventHandler) OpenChannel(ctx context.Context, iid string) interfa
 		chunkFileSizeBytes = defaultChunkFileSizeBytes
 	}
 	buildEventAccumulator := accumulator.NewBEValues(iid)
+	redactor := redact.NewStreamingRedactor(b.env)
```

Comment:
Will the redactor contain any state? If not, should we instantiate a single instance in NewBuildEventHandler?

Reply:
Yes, on each build event stream, the BuildStartedEvent contains a piece of BuildMetadata called ALLOW_ENV and we need to store that value in the redactor so it knows which pieces of the env need to be redacted (the env comes in a subsequent event)